Submitted By:            Fernando de Oliveira <famobr at yahoo dot com dot br>
Date:                    2014-06-05
Initial Package Version: 2.7.7
Upstream Status:         not applied
Origin:                  Upstream (partially)
URL:                     http://bugs.python.org/file17226/gdbopts.patch
Description:             Skip test_gdb, if python is built with optimizations

--- a/Lib/test/test_gdb.py.orig	2014-05-31 15:58:39.000000000 -0300
+++ b/Lib/test/test_gdb.py	2014-06-05 17:11:36.464091244 -0300
@@ -749,6 +749,15 @@
                                     r".*\na = 1\nb = 2\nc = 3\n.*")
 
 def test_main():
+    cflags = sysconfig.get_config_vars()['PY_CFLAGS']
+    final_opt = ""
+    for opt in cflags.split():
+        if opt.startswith('-O'):
+            final_opt = opt
+    if final_opt and final_opt != '-O0':
+        raise unittest.SkipTest("Python was built with compiler optimizations, "
+                                "tests can't reliably succeed")
+
     run_unittest(PrettyPrintTests,
                  PyListTests,
                  StackNavigationTests,
